lib/src/intl_suggestors/intl_migrator.dart

```diff
@@ -163,6 +170,20 @@ class IntlMigrator extends ComponentUsageMigrator {
     // }
   }
 
+  bool isLineIgnored(node) {
+    final ignoreComment = node.findPrevious(node.beginToken).precedingComments;
+    if (ignoreComment != null &&
+        ignoreComment.value().contains(ignoreStatement)) {
+      return true;
+    }
+    return false;
+  }
+
+  bool isFileIgnored(fileContents) {
```

Comment:
Not to prematurely optimize, but maybe we could pass in the file path and then remember the result for particular files.

Reply:
Well this is interesting. I went to look at how one might do this and came across lib/src/util/ignore_info.dart, which is doing basically the same thing, including finding the preceding comment, for the orcm_ignore and orcm_ignore_for_file comments.

And it's using, like, *all* the crazy Dart tricks at once. A static extension on CompilationUnit, a sync* method for finding the comments, and an Expando for remembering the ignore comment status. But I'm not sure we want to adapt or clone that code directly, but it might worth looking at for anything missed.

That part doesn't seem to do anything about skipping the entire file without visiting individual uses, but it does seem like there's an inherited 
  bool shouldSkip(FileContext context) => false;
which could possibly be overridden to check for the ignore_for_file comment.